tareas/Tarea_2_Andres/intervalos.jl

```diff
@@ -0,0 +1,352 @@
+	end
+	function intervalo_vacio(I::Intervalo{T}) where {T<:Real}   ### An 'Interval' arg
+		return Intervalo(T(NaN), T(NaN))
+	end
+end
+
+begin   ### Checking if an interval I is the empty interval
+	import Base: isempty
+	function isempty(I::Intervalo)
+		return isnan(I.infimo)
+	end
+end
+
+begin   ### [a,b] == [c,d] ⇒ a = b && b = d
+	import Base: ==
+	function ==(I1::Intervalo, I2::Intervalo)
+		if isempty(I1) && isempty(I2)   ### Empty interval cases
+			return true
+		else
+			inf1, inf2, sup1, sup2 = promote(I1.infimo, I2.infimo, I1.supremo, I2.supremo)
+			return (inf1 == inf2) && (sup1 == sup2)
+		end
+	end
+end
+
+begin   ### [a,b] ⊆ [c,d] ⇒ c ≤ a && b ≤ d
+	import Base: ⊆
+	function ⊆(I1::Intervalo, I2::Intervalo)
+		if isempty(I1)   ### Empty I case
+			return true
+		elseif (I2.infimo ≤ I1.infimo) && (I1.supremo ≤ I2.supremo)
+			return true
+		else
+			return false
+		end
+	end
+end
+
+begin   ### [a,b] ⊂ [c,d] ⇒ c ≤ a && b ≤ d   so that   (c = a && b = d) == false
+	function ⊂(I1::Intervalo, I2::Intervalo)
+		if isempty(I1)   ### Empty I case
+			return true
+		elseif (I2.infimo ≤ I1.infimo) && (I1.supremo ≤ I2.supremo)
+			if (I2.infimo == I1.infimo) && (I1.supremo == I2.supremo)
+				return false
+			else
+				return true
+			end
+		else
+			return false
+		end
+	end
+	function ⊃(I1::Intervalo, I2::Intervalo)
+		if isempty(I2)   ### Empty I case
+			return true
+		elseif (I1.infimo ≤ I2.infimo) && (I2.supremo ≤ I1.supremo)
+			if (I1.infimo == I2.infimo) && (I2.supremo == I1.supremo)
+				return false
+			else
+				return true
+			end
+		else
+			return false
+		end
+	end
+end   ### Note: There were no methods for this symbols
+
+begin   ### [a,b] ⪽ [c,d] ⇒ c < a && b < d
+	function isinterior(I1::Intervalo, I2::Intervalo)
+		if isempty(I1)   ### Empty I case
+			return true
+		elseif (I2.infimo < I1.infimo) && (I1.supremo < I2.supremo)
+			return true
+		else
+			return false
+		end
+	end
+	const ⪽ = isinterior
+end
+
+begin   ### x ∈ [a,b] ⇒ a ≤ x ≤ b
+	import Base: ∈
+	function ∈(x::Real, I::Intervalo)
+		return (I.infimo ≤ x) && (x ≤ I.supremo)
+	end
+end
+
+begin   ### [a,b] ⊔ [c,d] = [min(a,c), max(b,d)]
+	function hull(I1::Intervalo, I2::Intervalo)
+		if isempty(I1)   ### Empty interval cases
+			return I2
+		elseif isempty(I2)
+			return I1
+		elseif isempty(I1) && isempty(I2)
+			return I1
+		else
+			m = min(I1.infimo, I2.infimo)
+			s = max(I1.supremo, I2.supremo)
+			return Intervalo(m, s)
+		end
+	end
+	const ⊔ = hull
+end   ### Note: There were no methods for this symbol
+
+begin   ### [a,b] ∩ [c,d] ⇒ [max(a,c), min(b,d)]
+	import Base: ∩
+	function ∩(I1::Intervalo, I2::Intervalo)
+		if I1.infimo < I2.infimo && I1.supremo < I2.infimo   ### Disjoint intervals
+			return intervalo_vacio()
+		elseif I2.infimo < I1.infimo && I2.supremo < I1.infimo
+			return intervalo_vacio()
+		elseif isempty(I1)   ### Empty interval cases
+			return intervalo_vacio()
+		elseif isempty(I2)
+			return intervalo_vacio()
```

Comment:
```suggestion
		elseif isempty(I1)  ||  isempty(I2)
			return intervalo_vacio()
```

esto es más sencillo. si bien se podría tambien hacer un or con el caso anterior, la cuestión es que hacer puede provocar que la lectura del codigo sea más complicada, si bien puede quitar lineas repetidas para un mejor entendimiento yo uniria estas dos celdas porque es fácil leer la condición.